framework/CHANGELOG.md

```diff
@@ -12,6 +12,7 @@ Yii Framework 2 Change Log
 - Bug #15194: Fixed `yii\db\QueryBuilder::insert()` to preserve passed params when building a `INSERT INTO ... SELECT` query for MSSQL, PostgreSQL and SQLite (sergeymakinen)
 - Bug #15229: Fixed `yii\console\widgets\Table` default value for `getScreenWidth()`, when `Console::getScreenSize()` can't determine screen size (webleaf)
 - Bug #15234: Fixed `\yii\widgets\LinkPager` removed `tag` from `disabledListItemSubTagOptions` (SDKiller)
+- Enh #7988: Added `\yii\helpers\Html::errorSummary()` output formats: json and plain text (developeruz)
```

Comment:
It's a bit weird that plaintext and JSON formats are in HTML helper...

Reply:
How about putting them into console and json helpers?

Reply:
Yes, you are right 👍 I've added `\yii\helpers\Console::errorSummary()` and `\yii\helpers\Json::errorSummary()`. I moved duplicated part to the `collectErrors` method, but it is quite complicated and I have no idea how to make it simple. Overall it solved 3 codeclimate issues in `errorSummary` and moved complexity to `collectErrors` method